### PR TITLE
Added Doc for happy Number

### LINKED ISSUE #40: Happy Number documentation
## 📚 Documentation

(A clear and concise description of what the issue is.)
Add the article for Armstrong Number. The basic layout for the article should be:

What is Happy Number?
The explanation for Happy Number
Algorithmic Steps
Code (Use C++/Java and Python; At least two languages should be present and pick up code from NeoAlgo with apt credits to author)
Sample I/O
Time & Space Complexity
This article should be between 500-700 words in length. You should use traditional Markdown to prepare the content and the use of external images is prohibited. You can however create your own images. Run a proper Grammarly Lint check before submitting the PR.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

(Write your answer here.)
yes

### PR DESCRIPTION
**Pull request by nagasrihith608**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
yes
### Description
Added Doc for happy Number
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Fixed #40 
(Write your answer here.)

### CODE REVIEW

File: docs/mathematics/happy-number.mdx [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    # If the function returns True, it means number is a happy number
+    if(check == True):
+        print("The given number", number, "is a happy number!")
+    else:
+        print("The given number", number, "is not a happy number!")
+
+
+```
+
+</TabItem>
```
Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx:215]: Add a `</Tabs>` after `</TabItem>` and the CI would pass

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by nagasrihith608** [comment]
@HarshCasper, can you check and guide me if any changes required.

File: docs/mathematics/happy-number.mdx [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+## Introduction
+
+A number is called a happy number if it converges to 1 after a sequential operations where in each step we replace the orginal number by the sum of the squres of its individual digits.
+
+## Explanation
+
+When we have a digit given to us, we perform the operation, sum of squares of its individual
+digits but in some case we might arrive at the same number which we already came across. In such cases
+we can clearly see that it would never terminate so such numbers are not considered as happy numbers.
```
Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx]: Wrap the three lines into one.

Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx]: Can you rephrase this and use shorter and crisp statement?

File: docs/mathematics/happy-number.mdx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+3^2 + 7^2  = 58
+5^2 + 8^2  = 89
+8^2 + 9^2  = 145
+1^2 + 4^2 + 5^2 = 42
+4^2 + 2^2  = 20
+2^2 + 0^2  = 4
+```
+
+## Algorithm
+- Impliment a function which computes the sum of squares of the individual digits and returns the sum.
```
Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx:44]: It should be `Implement`

File: docs/mathematics/happy-number.mdx [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+- We assign 2 variables which keep track of the sum of squares in single iteration and two iterations 
+  using the above function implimented repeatedly.
```
Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx:46]: Don't break the line. Continue in the same line. Fix the typo for `implemented`

File: docs/mathematics/happy-number.mdx [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        print("The given number", number, "is not a happy number!")
+```
+
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+
+### Input:
```
Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx:215]: Use Heading-4

File: docs/mathematics/happy-number.mdx [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+
+
+```
+Enter a number to check if it's a happy number or not: 13
+```
+
+### Output:
```
Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx:222]: Use Heading-4

File: docs/mathematics/happy-number.mdx [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+
+```
+The given number 13 is a happy number!
+```
+### Input:
```
Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx:227]: Use Heading-4

File: docs/mathematics/happy-number.mdx [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+ 
+ 
+```
+Enter a number to check if it's a happy number or not: 4
+```
+ 
+### Output:
```
Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx:234]: Use Heading-4

File: sidebars.js [hunk 9]
```diff
@@ -64,7 +64,8 @@ module.exports = {
       items: [
         'mathematics/mathematics',
         'mathematics/armstrong-number',
-	  'mathematics/factorial-of-number'
+	'mathematics/factorial-of-number',
+        'mathematics/happy-number'
```
Comment (HarshCasper) [inline sidebars.js:68]: Indent and format it

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: sidebars.js (hunk 9 above)
Comment (nagasrihith608) [reply sidebars.js]: @HarshCasper actually, in my system, it is showing properly but here it is showing like this. Can you correct it as I am not able to if you can. Thank you.

File: docs/mathematics/happy-number.mdx [hunk 10]
```diff
@@ -0,0 +1,244 @@
+---
+title: Happy Number
+slug: /mathematics/happy-number
+---
+
+## Introduction
+
+A number is called a happy number if it converges to 1 after a sequential operation wherein each step we replace the original number by the sum of the squares of its individual digits.When we have a digit given to us, we perform the operation, the sum of squares of its individual digits but in some cases, we might arrive at the same number which we already came across. In such cases, we can clearly see that it would never terminate so such numbers are not considered as happy numbers.
+
```
Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx:9]: Give a space after full-stop

Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx:9]: Replace this: `When we have a digit given to us, we perform the operation, the sum of squares of its individual digits but in some cases, we might arrive at the same number which we already came across.`

with: 

`When a digit is inputted, we calculate the sum of squares of its individual digits. We find that in some cases, we arrive at the same number that we earlier came across during our calculation.`

File: docs/mathematics/happy-number.mdx [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+## Explanation
+
+If the number we calculate by the sum of squares appears again in the loop. In such a case, we don't see the loop terminating. So such a number is not a happy number.
```
Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx:12]: Replace this with: 

`If the number we are processing by calculating their sum of squares appears again in the loop, we will not see the loop terminating. Such a number cannot be considered as a happy number.`

File: docs/mathematics/happy-number.mdx [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+        number = number // 10;
+    
+    return sum_squares
+
+''' This function takes a number as input and checks
+    if the number is a happy number. The idea behind
+    the code is - we maintain two variables. The second
+    variable is one step ahead of the first one. If
+    the first variable becomes 1 so the second will be
+    1 anyhow, because 1*1 = 1. So, we terminate the loop.'''
```
Comment (HarshCasper) [inline docs/mathematics/happy-number.mdx:158]: Can you shorten this a bit to make it crisp?

**Review by HarshCasper — CHANGES_REQUESTED** [review]
This is the last set of review. Make the changes and I will send it for the peer review.

File: docs/mathematics/happy-number.mdx (hunk 12 above)
Comment (nagasrihith608) [reply docs/mathematics/happy-number.mdx:158]: actually, this is explained in the later code part so it will be good if I shorten it.

**Comment by nagasrihith608** [comment]
@HarshCasper, I just changed these can you ask a peer to review it if it's ok. Thank you!

**Review by HarshCasper — APPROVED** [review]

**Review by ankitaggarwal23 — APPROVED** [review]
